File: eTour/src/main/java/com/example/service/CustomerService.java
package com.example.service;

import com.example.model.Customer;
import com.example.repository.CustomerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CustomerService {
    
    @Autowired
    private CustomerRepository customerRepository;

    public Customer createCustomer(Customer customer) {
        return customerRepository.save(customer);
    }

    public Optional<Customer> getCustomerById(Integer customerId) {
        return customerRepository.findById(customerId);
    }

    public List<Customer> getAllCustomers() {
        return customerRepository.findAll();
    }

    public Customer updateCustomer(Customer customer) {
        return customerRepository.save(customer);
    }

    public void deleteCustomer(Integer customerId) {
        customerRepository.deleteById(customerId);
    }
    
    
    public boolean validateCustomer(String email, String password) {
        Customer customer = customerRepository.getUserByEmailIdAndPass(email, password);
        return customer != null;
    }
}

File: Service/CustomerService.cs
using eTour.Model;
using eTour.Repository;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace eTour.Service
{
    public class CustomerService : ICustomerService
    {
        private readonly Appdbcontext context;

        public CustomerService(Appdbcontext context)
        {
            this.context = context;
        }
        public async Task<ActionResult<Customer>> CreateCustomer(Customer customer)
        {
            context.Customers.Add(customer);
            await context.SaveChangesAsync();
            return customer;
        }

        public async Task<Customer> DeleteCustomer(int customer_id)
        {
            Customer customer = context.Customers.Find(customer_id);
            if (customer != null)
            {
                context.Customers.Remove(customer);
                await context.SaveChangesAsync();
            }
            return customer;
        }

        public async Task<ActionResult<IEnumerable<Customer>>> GetAllCustomer()
        {
            return await context.Customers.ToListAsync();
        }

        public async Task<ActionResult<Customer>?> GetCustomerById(int customer_id)
        {
            var customer = await context.Customers.FindAsync(customer_id);
            if (customer == null)
            {
                return null;
            }
            return customer;
        }

        public async Task<ActionResult<Customer>> UpdateCustomer(int customer_id, Customer customer)
        {
            if (customer_id != customer.Customer_Id)
            {
                return null;
            }
            context.Entry(customer).State = EntityState.Modified;

            try
            {
                await context.SaveChangesAsync();
            }
            catch
            {
                Console.WriteLine("Not Updated");
            }

            return customer;
        }

        public bool ValidateCustomer(string email, string password)
        {
            var customer = context.Customers
                               .FirstOrDefault(c => c.Customer_EmailId == email);

            
            return customer != null && customer.Customer_Password == password;

        }
    }
}
